### PR TITLE
add configurable furnace, crafting table and third-party mod support + configuration

### LINKED ISSUE #3: feature: ability to configure which screens are paused
Currently only instances of "AbstractInventoryScreen" are paused. Is it possible to enable users to provide their own list of class names?

### PR DESCRIPTION
**Pull request by macbrayne**
Fixes #3. This isn't compatible with all third-party mods because some don't use the HandledScreen super class and implement Screen by themselves.

- adds support for furnaces, crafting tables and a limited set of third-party mods.
- adds a config screen with mod menu integration

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -14,6 +14,23 @@ dependencies {
 	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
 	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
 	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
+
+	// Mod Menu, to add the hook for a config screen
+	modImplementation "io.github.prospector:modmenu:${project.modmenu_version}"
```
Comment (umollu) [inline build.gradle:24]: Change modmenu maven: check https://github.com/umollu/ash/blob/master/build.gradle

File: build.gradle [hunk 2]
```diff
@@ -14,6 +14,23 @@ dependencies {
...
+
+	modApi("me.shedaniel.cloth:config-2:${project.cloth_version}") {
+		exclude(group: "net.fabricmc.fabric-api")
+	}
+	include("me.shedaniel.cloth:config-2:${project.cloth_version}") {
+		exclude group: "net.fabricmc.fabric-api", module: "fabric-api"
+	}
+
+	modApi("me.sargunvohra.mcmods:autoconfig1u:${project.auto_config_version}") {
+		exclude group: "net.fabricmc.fabric-api", module: "fabric-api"
```
Comment (umollu) [inline build.gradle]: you can remove the exclude for autoconfig1u

File: build.gradle [hunk 3]
```diff
@@ -14,6 +14,23 @@ dependencies {
...
+	}
+	include("me.sargunvohra.mcmods:autoconfig1u:${project.auto_config_version}") {
+		exclude group: "net.fabricmc.fabric-api", module: "fabric-api"
```
Comment (umollu) [inline build.gradle:30]: you can remove the exclude for autoconfig1u

**Comment by macbrayne** [comment]
I've now also updated the yarn mappings and the config library

**Review by umollu — APPROVED** [review]

**Pull request merged**